[WIP] Describe "System Configuration" fields - draft (#4830)
jenkins-infra/jenkins.io: content/doc/book/managing/system-configuration.adoc

:toc: left
endif::[]

= Configure System

> Use the *Manage Jenkins >> System Configuration* page to configure
> many standard aspects of your Jenkins server, including:

Reviewer: ```suggestion
Use the *Manage Jenkins >> System Configuration* page to configure many standard aspects of your Jenkins server, including:
```
full stop for each sentence (1 per line) makes it easier to review

** JDK installations
** Build tools - Ant and Maven installations
** Version control tools
** Email configuration

This page discusses the fields that are present after a standard Jenkins installation using the "Suggested plugins".
Plugins installed on your system may add sections to this page.

You can set these configurations using the Jenkins dashboard as described here or you can set them using Configuration as Code, where you define the configuration you want in a YAML file that you then deploy to a server to configure Jenkins.

> See link:http://localhost:4242/doc/book/managing/casc/[Configuration as Code]
> for more information.

Reviewer: ```suggestion
Refer to link:http://localhost:4242/doc/book/managing/casc/[Configuration as Code] for more information.
```
update syntax/full stop per line

== General system-wide configurations

> The top of the System Configuration page
> sets basic information about the controller:

Reviewer: ```suggestion
The top of the System Configuration page sets basic information about the controller:
```
same as above, full stop per line

> * *Home Directory* -- Display the location of the Jenkins home directory.
> You can change this location
> by modifying the value of the `JENKINS_HOME` environment variable.
> 
> * *System Message* -- Text that is displayed
> at the top of your Jenkins home page.
> Use this page to display the name of the server
> and a short description of its purpose.
> It is also used to display announcements about scheduled down time
> and other information of interest to all Jenkins users.
> This field can include HTML tags
> or a pointer to a Wiki page and information about who to contact.
> 
> Inline help is available for most fields on most *Manage Jenkins* pages.
> To access the help, select the *?* icon to the right of each field.
> Click the *?* icon again to hide the help text.

Reviewer: ```suggestion
* *Home Directory* - Display the location of the Jenkins home directory.
** You can change this location by modifying the value of the `JENKINS_HOME` environment variable.

* *System Message* - Text that is displayed at the top of your Jenkins home page.
** Use this page to display the name of the server and a short description of its purpose.
It is also used to display announcements about scheduled down time and other information of interest to all Jenkins users.
This field can include HTML tags or a pointer to a Wiki page and information about who to contact.

Inline help is available for most fields on most *Manage Jenkins* pages.
To access the help, select the *?* icon to the right of each field name.
Select the *?* icon again to hide the help text.
```
update some syntax/formatting (full stops)

== Executors on controller

The *# of executors* field configures the number of executors on the controller.
Typically, this should be set to 0 to prevent builds from running on the controller.
We will discuss executors for nodes and agents in a later section.

== Quiet period

This field defines an interval to wait before actually starting a build.
This can be used to collapse multiple adjacent commits into one build.

> ////
> These fields no longer appear on the *System Configuration* page.
> == SCM tools
> 
> This field configures tools on the controller itself;
> it does not control the tools that are configured in a build agent
> for running builds.
> 
> Git is supported out-of-the-box.
> Plugin support is provided for
> Subversion, Mercurial, Perforce, Team Foundation Server, and other SCMs.
> In most cases, the plugin for the tool must be installed separately
> before you configure the tool on the System Configuration page.
> 
> == SCM and authentication
> 
> Most SCM tools support or require authentication when talking to servers.
> Different SCMs are configured differently,
> but most (including Git, SVN, Mercurial, and Team Foundation Server)
> use the Credentials plugin.
> However, Perforce requires that each project
> enters credentials separately.
> ////

Reviewer: ```suggestion

```
remove text if these are no longer displayed

== Jenkins location

> You must configure the **Jenkins Location** area
> to identify this particular instance of Jenkins.
> 
> Two items must be defined here:
> 
> * *Jenkins URL* -- Replace the `localhost:5000` string with information
> about the actual location people can use to access this Jenkins instance.
> This information can be:
> 
> ** The actual URL that people use to access Jenkins.
> ** The host alias as defined by the name service.
> If your IT people cannot provide this, use dynamic DNS.
> This makes your instance easier for your users to access
> and makes your service relocatable.

Reviewer: ```suggestion
You must configure the **Jenkins Location** area to identify this particular instance of Jenkins.

Two items must be defined here:

* *Jenkins URL* -- Replace the `localhost:5000` string with information about the actual location people can use to access this Jenkins instance.
This information can be:

** The actual URL that people use to access Jenkins.
** The host alias as defined by the name service.
If your IT people cannot provide this, use dynamic DNS.
This makes your instance easier for your users to access, and makes your service relocatable.
```
small suggestion for formatting/syntax

> [NOTE]
> ====
> *Invest in a good URL!*
> If your users cannot see Jenkins, much of the benefit is lost.
> Make the URL easy to  remember.  For example:
> 
> * Bad: `\http://tjek-02.int.rev.example.org`
> * Good: `\https://jenkins.example.org`
> ====

Reviewer: ```suggestion
[NOTE]
====
*Invest in a good URL!*
If your users cannot see Jenkins, much of the benefit is lost.
Make the URL easy to  remember. 
For example:

* Bad: `\http://tjek-02.int.rev.example.org`
* Good: `\https://jenkins.example.org`
====
```
small formatting suggestion

> * *System Admin e-mail address* -- Define the email address to which email
> about general Jenkins issues are sent.
> This can be an individual's email address
> or an email alias that contains multiple email addresses.
> 
> == Timestamper
> 
> The **Timestamper** fields define the format used
> to display timestamp information.

Reviewer: ```suggestion
* *System Admin e-mail address* -- Define the email address to which email about general Jenkins issues are sent.
This can be an individual's email address or an email alias that contains multiple email addresses.

== Timestamper

The **Timestamper** fields define the format used to display timestamp information.
```
small formatting suggestion

== More about system configuration

Jenkins is very flexible and powerful.
Default configurations can be used for most fields while you are getting started.

> As you learn more about how Jenkins works and how your project uses Jenkins,
> you can modify the values of settings on this page.
> Use these settings to fine-tune the behavior of your Jenkins instance.

Reviewer: ```suggestion
As you learn more about how Jenkins works and how your project uses Jenkins, you can modify the values of settings on this page.
Use these settings to fine-tune the behavior of your Jenkins instance.
```
small formatting suggestion

> When you are done modifying this screen,
> scroll to the bottom and click
> *Save* to apply your configuration changes.

Reviewer: ```suggestion
When you are done modifying this screen, scroll to the bottom and select *Save* to apply your configuration changes.
```
small formatting and syntax suggestion

== Management options

> Select **Managment Options** to start, stop, and reload the Jenkins service.
> 
> To reload the configuration from disk,
> Jenkins rereads the configuration from `$JENKINS_HOME`
> without restarting Jenkins.
> 
> The **Prepare for Shutdown** option
> prevents new jobs from being started
> and waits for running jobs to complete.
> Use this option to ensure a clean job termination before you shut down or restart Jenkins.
> 
> The Management page provides Jenkins Upgrade management.
> If you are running a standalone server,
> you can perform a one-click upgrade.
> Otherwise, you have a direct link to the latest WAR.
> 
> **Manage Old Data** helps to handle dropped configuration items after an upgrade.
> 
> ////
> == What did we learn?
> ////

Reviewer: ```suggestion
Select **Managment Options** to start, stop, and reload the Jenkins service.

To reload the configuration from disk,
Jenkins rereads the configuration from `$JENKINS_HOME`
without restarting Jenkins.

The *Prepare for Shutdown* option prevents new jobs from being started, and waits for running jobs to complete.
Use this option to ensure a clean job termination before you shut down or restart Jenkins.

The Management page provides Jenkins Upgrade management.
If you are running a standalone server, you can perform a one-click upgrade.
Otherwise, you have a direct link to the latest WAR.

*Manage Old Data* helps to handle dropped configuration items after an upgrade.
```
small formatting and syntax suggestion, remove unneeded text
